portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,21 +35,38 @@
 public class DataServlet extends HttpServlet {
   private List<String> comments;
   private Gson gson;
+  private DatastoreService datastore;
   @Override
   public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
     gson = new Gson();
-    comments = new ArrayList<>();
+    
+    
   }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numComments = getNumParameter(request, "numComments");
+    comments = new ArrayList<>(numComments);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity comment : results.asIterable()) {
+      if (comments.size() == numComments) {
+        break;
+      }
+      comments.add((String) comment.getProperty("text"));
+    }
     String jsonComments = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(jsonComments);
   }
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = getParameter(request, "commentText");
-    comments.add(comment);
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("timestamp",timestamp);
```

Comment:
May as well just inline this to commentEntity.setProperty("timestamp", System.currentTimeMillis());

Reply:
Will fix into inline